### PR TITLE
Server emits a telemetry event for initialization

### LINKED ISSUE #430: Add telemetry/event support
collect at least the startup event with 
- Server Version
- JVM Version
- memory

### PR DESCRIPTION
**Pull request by datho7561**
Please refer to `USAGE_DATA.md` for a full list of data collected.

Add `xml.telemetry.enable` to control if telemetry events are sent.

Closes #430

Rebase of #493

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: USAGE_DATA.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+ * JVM information:
+    * Whether LemMinX is being run with Java or as a GraalVM native image (binary)
+    * The name of the vm (`java.vm.name`)
+    * The name of the runtime (`java.runtime.name`)
+    * The version of the JVM (`java.version`)
+    * The free, total, and max VM memory
+    * Does NOT include the `JAVA_HOME` environment variable for privacy reasons
+ * Version information:
+    * The Maven version number
```
Comment (fbricon) [inline USAGE_DATA.md:16]: Maven??

File: USAGE_DATA.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    * The branch of the build
+    * The short form of the commit id
+    * The short form of the commit message
```
Comment (fbricon) [inline USAGE_DATA.md:19]: I don't think those bring us any actionable data, we can skip them

File: USAGE_DATA.md (hunk 1 above)
Comment (fbricon) [reply USAGE_DATA.md:16]: The server version

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/TelemetryManager.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * Telemetry manager.
+ *
+ * @author Angelo ZERR
+ */
+public class TelemetryManager {
+
+	/**
+	 * "startup" telemetry event name
+	 */
+	private static final String STARTUP_EVENT_NAME = "startup";
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/TelemetryManager.java:25]: server_startup or server_initialized

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/TelemetryManager.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			telemetryEvent(STARTUP_EVENT_NAME, new InitializationTelemetryInfo());
+		}
+	}
+
+	/**
+	 * Send a telemetry event on call of a service.
+	 *
+	 * @param serviceName service name.
+	 */
+	public void onServiceCall(String serviceName) {
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/TelemetryManager.java:64]: seems unused

Comment (datho7561) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/TelemetryManager.java:64]: Yes, this is left over from the old PR. I'll remove it.

File: USAGE_DATA.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+When telemetry events are enabled, the following information is emitted when the language server starts:
+
+ * JVM information:
+    * Whether LemMinX is being run with Java or as a GraalVM native image (binary)
+    * The name of the vm (`java.vm.name`)
+    * The name of the runtime (`java.runtime.name`)
+    * The version of the JVM (`java.version`)
+    * The free, total, and max VM memory
+    * Does NOT include the `JAVA_HOME` environment variable for privacy reasons
```
Comment (rgrunber) [inline USAGE_DATA.md:14]: I would list this outside of "JVM information". Maybe with "Note : Does NOT include ..."

**Review by rgrunber — APPROVED** [review]
Just a few small things to address, but otherwise I think we're ready to merge this.

**Pull request merged**